overlord/hookstate/ctlcmd/refresh.go

```diff
@@ -155,10 +155,25 @@ func getUpdateDetails(context *hookstate.Context) (*updateDetails, error) {
 		return nil, err
 	}
 
-	// TODO: pending
+	var snapst snapstate.SnapState
+	if err := snapstate.Get(st, context.InstanceName(), &snapst); err != nil {
+		return nil, fmt.Errorf("internal error: cannot get snap state for %q: %v", context.InstanceName(), err)
+	}
+
+	var pending string
+	switch {
+	case snapst.RefreshInhibitedTime != nil:
```

Comment:
this check here will be correct only if we move where we do softCheckNothingRunningForRefresh because now that is done late (for our new POV) in doInstall

Reply:
I'm addressing this in a separate PR that deals with locking in Before/Done/Error handlers of the hook; I think this PR isn't directly affected and could land?